src/components/AvatarWithImagePicker.js

```diff
@@ -293,92 +245,91 @@ class AvatarWithImagePicker extends React.Component {
                                                 // In order for the file picker to open dynamically, the click
                                                 // function must be called from within a event handler that was initiated
                                                 // by the user on Safari.
                                                 if (index === 0 && Browser.isSafari()) {
                                                     openPicker({
-                                                        onPicked: this.showAvatarCropModal,
+                                                        onPicked: showAvatarCropModal,
                                                     });
                                                 }
                                             }}
                                             menuItems={menuItems}
-                                            anchorPosition={this.props.anchorPosition}
+                                            anchorPosition={props.anchorPosition}
                                             withoutOverlay
-                                            anchorRef={this.anchorRef}
-                                            anchorAlignment={this.props.anchorAlignment}
+                                            anchorRef={anchorRef}
+                                            anchorAlignment={props.anchorAlignment}
                                         />
                                     );
                                 }}
                             </AttachmentPicker>
                         )}
                     </AttachmentModal>
                 </View>
-                {this.state.validationError && (
+                {validationError && (
                     <DotIndicatorMessage
                         style={[styles.mt6]}
-                        messages={{0: this.props.translate(this.state.validationError, this.state.phraseParam)}}
+                        messages={{0: props.translate(validationError, phraseParam)}}
                         type="error"
                     />
                 )}
                 <AvatarCropModal
-                    onClose={this.hideAvatarCropModal}
-                    isVisible={this.state.isAvatarCropModalOpen}
-                    onSave={this.props.onImageSelected}
-                    imageUri={this.state.imageUri}
-                    imageName={this.state.imageName}
-                    imageType={this.state.imageType}
-                    maskImage={this.props.editorMaskImage}
+                    onClose={hideAvatarCropModal}
+                    isVisible={isAvatarCropModalOpen}
+                    onSave={props.onImageSelected}
+                    imageUri={imageUri}
+                    imageName={imageName}
+                    imageType={imageType}
+                    maskImage={props.editorMaskImage}
                 />
-            </View>
-        );
-    }
+        </View>
+    );
```

Comment:
replace the `withLocalize` to `useLocalize` hook
add the `displayName` [prop](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#function-component-style).